### PR TITLE
Fixed spotbugs for azure-common-mgmt.

### PR DESCRIPTION
**Pull request by sima-zhu**
Fixed spotbugs for azure-common-mgmt. pkg.

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml [hunk 1]
```diff
@@ -195,4 +195,21 @@
...
+      <!-- The poll strategy class has no necessary to serialize or deserialize. -->
+      <Match>
+        <Class name="~com\.azure\.common\.mgmt\.(CompletedPollStrategy\$CompletedPollStrategyData|ProvisioningStatePollStrategy\$ProvisioningStatePollStrategyData)"/>
+        <Bug pattern="SE_NO_SERIALVERSIONID"/>
+      </Match>
+
+      <Match>
+        <Class name="~com\.azure\.common\.mgmt\.(CompletedPollStrategy\$CompletedPollStrategyData|ProvisioningStatePollStrategy\$ProvisioningStatePollStrategyData)"/>
+        <Bug pattern="SE_BAD_FIELD"/>
+      </Match>
```
Comment (JonathanGiles) [inline eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: Can you leave a comment here about why these checks are suppressed?

Comment (sima-zhu) [reply eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: Added reasonable comments.

Comment (JonathanGiles) [reply eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: Did you confirm with @anuchandy or @jianghaolu about these classes and their need to be serializable? I note that CompletedPollStrategy has a method `strategyData()` that returns `Serializable`.

Comment (sima-zhu) [reply eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: Will confirm with Anu. Jianghao said it is ok to his adjudgment, but needs confirm from @anuchandy . I will ask him to take a look as well.

Comment (anuchandy) [reply eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: @sima-zhu @JonathanGiles, these types needs to be serializable. The use case is - 

User can start a long running operation and operations's progress can be tracked by polling. In such cases, we want to give user the ability to serialize the polling data (into DB e.g.) and deserialize it and use it to  continue the polling. You can imagine the case where application is running in a state-less micro-service. If an instance goes down another instance might need to continue from where the first instance stopped. Support for polling data serialization enables this.

Comment (JonathanGiles) [reply eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: @sima-zhu Please make the changes requested by @anuchandy - thanks.

**Pull request merged**